plot_krun_results.py

```diff
@@ -85,74 +86,64 @@ def main(is_interactive, data_dcts, window_size, outfile, xlimits, single_exec,
+                    unique = data_dcts[machine]['unique_outliers'][key]
+                    common = data_dcts[machine]['common_outliers'][key]
+                else:
+                    unique, common = None, None
+                if is_interactive:
                     draw_page(is_interactive, key, executions, machine,
                               window_size, xlimits, single_exec, outliers,
                               unique, common, mean, sigma)
-    else:  # Generate all charts and write to a single PDF.
-        from matplotlib.backends.backend_pdf import PdfPages
-        with PdfPages(outfile) as pdf:
-            set_pdf_metadata(pdf)
-            for machine in data_dcts:
-                keys = sorted(data_dcts[machine]['data'].keys())
-                for key in keys:
-                    bench, vm, variant = key.split(':')
-                    executions = data_dcts[machine]['data'][key]
-                    if benchmark is None and len(executions) == 0:
-                        print('Skipping: %s (no executions)' % key)
-                    elif benchmark is None and len(executions[0]) == 0:
-                        print('Skipping: %s (benchmark crashed)' % key)
-                    elif (benchmark is not None) and key != benchmark:
-                        continue
-                    else:
-                        if with_outliers:
-                            outliers = data_dcts[machine]['outliers'][key]
-                        else:
-                            outliers = None
-                        if unique_outliers:
-                            unique = data_dcts[machine]['unique_outliers'][key]
-                            common = data_dcts[machine]['common_outliers'][key]
-                        else:
-                            unique, common = None, None
-                        fig, export_size = draw_page(is_interactive,
-                                                     key,
-                                                     executions,
-                                                     machine,
-                                                     window_size,
-                                                     xlimits,
-                                                     single_exec,
-                                                     outliers,
-                                                     unique,
-                                                     common,
-                                                     mean,
-                                                     sigma)
-                        fig.set_size_inches(*export_size)
-                        pdf.savefig(fig,
-                                    dpi=fig.dpi, orientation='landscape',
-                                    bbox_inches='tight')
-                        plt.close()
-        print('Saved: %s' % outfile)
+                else:
+                    fig, export_size = draw_page(is_interactive,
```

Comment:
Do we need the call to drawpage() twice? Perhaps we can always assign the return values, but only use them in one mode?